src/information.jl

```diff
@@ -0,0 +1,66 @@
+# This file is a part of MGVInference.jl, licensed under the MIT License (MIT).
+
+function fisher_information(dist::Normal)
+    μval, σval = params(dist)
+    res = [1/σval^2, 1/2 * 1/σval^4]
+    PDiagMat(res)
+end
+
+function fisher_information(dist::MvNormal)
+    μval, σval = params(dist)
+    invσ = inv(σval)
+
+    μdof, σdof = length(μval), length(σval)
+    dof = μdof + σdof
+    res = spzeros(dof, dof)
+
+    res[1:μdof, 1:μdof] = invσ
+
+    for i in 1:σdof
+        for j in 1:σdof
+            full_i, full_j = μdof + i, μdof + j
+
+            # translation from flat index of matrix to row/col
+            i_row, i_col = (i-1) ÷ μdof + 1, (i-1) % μdof + 1
+            j_row, j_col = (j-1) ÷ μdof + 1, (j-1) % μdof + 1
+
+            res[full_i, full_j] = invσ[j_col, i_row] * invσ[i_col, j_row] / 2
+        end
+    end
+
+    PDSparseMat(res)
+end
+
+function fisher_information(dist::Exponential)
+    λ = params(dist)[1]
+    res = [1/λ^2,]
+    PDiagMat(res)
+end
+
+function fisher_information(dist::Product)
+    dists = dist.v
+    λinformations = map(fisher_information, dists)
+    λinformations |> BlockDiagonal |> sparse |> PDSparseMat
+end
+
+function λ_fisher_information(f::Function, p::Vector)
+    dists = values(f(p))
+    posdef_map = m -> LinearMap(m, isposdef=true)
```

Comment:
We should use `LinearMap(m::AbstractPDMat, issymmetric = true, ishermitian = true, isposdef = true)` (see JuliaStats/PDMats.jl#121)